src/directives/ng-file-drop.ts

```diff
@@ -105,7 +105,7 @@ export class NgFileDropDirective implements OnChanges, OnInit {
     }
 
     if (this.options.filterExtensions && this.options.allowedExtensions && this.files && this.files.length) {
-      this.files = this.files.filter(f => {
+      this.files = [].filter.call(this.files, (f: any) => {
```

Comment:
Ok, so I'll have to change the array type also. Will commit in a couple minutes.